### ISSUE TITLE
Wrap language constructs which can not be used as expressions

### ISSUE THREAD

**Issue by gchallen** [pinterest/ktlint#1097]
## Expected Behavior

`ktlint` should not pass code like this:

```kotlin
class Course(var number: String) { fun changeNumber(newNumber: String) { number = newNumber } }
```
which is much clearer written:

```kotlin
class Course(var number: String) {
  fun changeNumber(newNumber: String) {
    number = newNumber
  }
}
```
(Yes, I know that this is a dumb example. But you get the drift.)

## Current Behavior

At least with the ruleset that I have enabled the code above is accepted without errors, until you hit the line limit if there is one.

I'm curious about whether this is on the roadmap or considered out of scope for `ktlint`.

## Additional information
* Current version of ktlint: 0.40.0

**Comment by paul-dingemans** [comment]
This rule is stated to generic to be picked up. But at minimum, wrapping has to be enforced in following cases:
* package
* import
* type_alias
* classes
* functions
* control loops which can not be used as expression:
  * for loop
  * while
  * do-while

**Comment by paul-dingemans** [comment]
> ##
> `ktlint` should not pass code like this:
> 
> ```kotlin
> class Course(var number: String) { fun changeNumber(newNumber: String) { number = newNumber } }
> ```

This is mostly covered by #2001. Other constructs can only be placed on same line when separated by `;` which will be covered by #1078. Issue is closed as duplicate.
